### ISSUE TITLE
TypeScript issue on display prop

### ISSUE THREAD

**Issue by Adam-Schlichtmann** [mmazzarolo/react-native-modal-datetime-picker#567]
## Environment

[Expo Snack](https://snack.expo.io/@mada1113/date-picker-ts-issue)

## Platforms

iOS - IPad Air 3

## Versions

- Android: N/A
- iOS: 14.4.2
- react-native-modal-datetime-picker: 9.2.3
- [react-native-community/datetimepicker](https://github.com/react-native-community/datetimepicker): 3.2.0
- react-native: 63 (via Expo v41)
- react: 16.3.1

## Description

Display prop is giving a TS error when set to inline or compact
EDIT: This is also an issue for setting the display prop to calendar or clock (Anything that isn't spinner or default)

![Screen Shot 2021-05-10 at 11 58 30 AM](https://user-images.githubusercontent.com/35603662/117697062-aad94880-b187-11eb-8a3e-fa5f6698aab0.png)

## Reproducible Demo

Example using both packages available [Expo Snack](https://snack.expo.io/@mada1113/date-picker-ts-issue)

**Comment by DilipCoder** [comment]
hi,
I would like to work on this issue.
can you please break down tasks for me?

**Comment by mmazzarolo** [comment]
Hey @DilipCoder , I'd love to give you more details but I'm not too familiar with this part of TypeScript. Basically, we'd need to differentiate props depending on what Platform.OS is. Not sure how other repos are handling this

**Comment by DilipCoder** [comment]
okay I am forking the repository will raise a PR this weekend

**Comment by mnajjarian** [comment]
Hey @DilipCoder, How does it go? I can do it if you don't mind.

**Comment by DilipCoder** [comment]
hey i was going through the design patterns used in this repo. I will raise the draft pr soon.

**Comment by toothlesspanda** [comment]
Hi!
So this is weird, I am still having the same error as the primary issue here
![image](https://user-images.githubusercontent.com/6379428/145307044-ff264311-f166-4228-bfd0-75afeb2ee05e.png)
I was trying to follow up the progress of this, then I saw this issue was solved in between some versions (I think version 11), but I am still getting the same error and the version of my package is 13, which is the latest release. 

```
 "@react-native-community/datetimepicker": "^4.0.1",
 "react-native-modal-datetime-picker": "^13.0.0",
```

**Comment by mmazzarolo** [comment]
@toothlesspanda the issue has not been solved :) We're waiting for contributions in #585

**Comment by matospanda** [comment]
@mmazzarolo I applied your request changes in that PR, I will commit later 🙏🏾

**Comment by mmazzarolo** [comment]
thank you!

**Comment by mmazzarolo** [comment]
closed #636
